Domains/Jobs/Job.php

```diff
@@ -0,0 +1,246 @@
+
+    /**
+     * Model's timestamp fields
+     *
+     * @var bool
+     */
+    public $timestamps = false;
+
+    /**
+     * Following attributes could be mass assignable
+     *
+     * @var string[]
+     */
+    protected $fillable = [
+        'attempts',
+        'payload',
+        'queue',
+        'reserved_at',
+    ];
+
+    /**
+     * Following attributes will be hidden from arrays
+     *
+     * @var string[]
+     */
+    protected $hidden = [
+        'available_at',
+        'created_at',
+        'reserved_at',
+    ];
+
+    /**
+     * Casting attributes to their native types
+     *
+     * @var string[]
+     */
+    protected $casts = [
+        'attempts' => 'int',
+        'available_at' => 'datetime',
+        'created_at' => 'datetime',
+        'payload' => 'array',
+        'queue' => 'string',
+        'reserved_at' => 'datetime',
+    ];
+
+    /**
+     * Instantiate the Job model
+     */
+    public function __construct(array $attributes = [])
+    {
+        parent::__construct($attributes);
+
+        $this->setDefaultQueue(Config::getVar('queues', 'default_queue', null));
+        $this->setMaxAttempts(self::DEFAULT_MAX_ATTEMPTS);
+    }
+
+    /**
+     * Set the default queue
+     *
+     * @param string|mixed $value
+     *
+     */
+    public function setDefaultQueue(?string $value): self
+    {
+        $this->defaultQueue = $value;
+
+        return $this;
+    }
+
+    /**
+     * Get Queue name, in case of nullable value, will be the default one
+     *
+     *
+     */
+    public function getQueue(?string $queue): string
+    {
+        return $queue ?: $this->defaultQueue;
+    }
+
+    /**
+     * Set the Job's max attempts
+     *
+     *
+     */
+    public function setMaxAttempts(int $maxAttempts): self
+    {
+        $this->maxAttempts = $maxAttempts;
+
+        return $this;
+    }
+
+    /**
+     * Get the Job's max attempts
+     *
+     */
+    public function getMaxAttempts(): int
+    {
+        return $this->maxAttempts;
+    }
+
+    /**
+     * Add a local scope for not excedeed attempts
+     *
+     *
+     */
+    public function scopeNotExcedeedAttempts(Builder $query): Builder
+    {
+        return $query->where('attempts', '<', $this->getMaxAttempts());
+    }
+
+    /**
+     * Add a local scope to handle jobs associated in a queue
+     *
+     *
+     */
+    public function scopeQueuedAt(
+        Builder $query,
+        ?string $queue = null
+    ): Builder {
+        return $query->where('queue', $this->getQueue($queue));
+    }
+
+    /**
+     * Retrieve available jobs
+     *
+     *
+     */
+    public function scopeIsAvailable(Builder $query): Builder
+    {
+        return $query->whereNull('reserved_at')
+            ->where('available_at', '<=', $this->currentTime());
+    }
+
+    /**
+     * Get queue's size
+     *
+     *
+     */
+    public function size(?string $queue = null): int
+    {
+        return $this->queuedat($this->getQueue($queue))
```

Comment:
(Should this be `queuedAt`? Not sure where this is implemented, suspecting a `Model` convention.)